mtp_api/apps/core/management/commands/delete_old_data.py

```diff
@@ -0,0 +1,205 @@
+import datetime
+import textwrap
+from typing import Sequence
+
+from django.core.management import BaseCommand
+from django.db import transaction
+from django.utils import timezone
+
+from credit.models import Credit
+from disbursement.models import Disbursement
+from notification.models import CreditEvent, DisbursementEvent, PrisonerProfileEvent, \
+    RecipientProfileEvent, SenderProfileEvent
+from payment.models import BillingAddress, Payment
+from security.models import (
+    BankAccount, RecipientProfile, SenderProfile, PrisonerProfile, SavedSearch
+)
+from transaction.models import Transaction
+from user_event_log.models import UserEvent
+
+
+class Command(BaseCommand):
+    """
+    Deletes data which is older than 7 years
+    """
+    help = textwrap.dedent(__doc__).strip()
+
+    def print_message(self, message, depth=0):
+        self.write('\t' * depth + message)
+
+    def handle(self, *args, **options):
+        verbosity = options['verbosity']
+
+        self.write = self.stdout.write if verbosity else lambda m: m
+
+        today = timezone.localdate()
+        seven_years_ago = today - datetime.timedelta(days=7*365)
+
+        self.print_message(f'\nDeleting Credits older than {seven_years_ago}...')
+        credits_to_delete = Credit.objects.filter(created__lt=seven_years_ago)
+        for credit in credits_to_delete:
+            self.delete_credit(credit)
+
+        self.print_message(f'\nDeleting Disbursments older than {seven_years_ago}...')
+        disbursements_to_delete = Disbursement.objects.filter(created__lt=seven_years_ago)
+        for disbursement in disbursements_to_delete:
+            self.delete_disbursement(disbursement)
+
+        self.print_message(f'\nDeleting Transactions older than {seven_years_ago}...')
+        records_deleted = Transaction.objects.filter(received_at__lt=seven_years_ago).delete()
+        self.print_message(f'Records deleted: {records_deleted}.', 1)
+
+        self.print_message(f'\nDeleting Payment older than {seven_years_ago}...')
+        records_deleted = Payment.objects.filter(modified__lt=seven_years_ago).delete()
```

Comment:
why does this one use modified date to filter and so should the others use it too instead of created?

Reply:
Some tables have indexes on different columns, `payment_payment` has an index on `modified` so this takes advantage of that. As discussed some time ago it shouldn't be far off the `created` date anyway (usually minutes/hour as the payment is taken)